src/vsphere_cpi/lib/cloud/vsphere/attach_tag_to_vm.rb

```diff
@@ -0,0 +1,243 @@
+require 'cloud/vsphere/logger'
+require 'vsphere-automation-cis'
+require 'vsphere-automation-vcenter'
+
+module  VSphereCloud
+  module TaggingTag
+
+    class VCenterBadEntityError < StandardError
+      def initialize(name, type)
+        @name = name
+        @type = type
+      end
+
+      def to_s
+        "Unable to locate #{@type} with name '#{@name}' on vCenter, skip attaching this #{@type}."
+      end
+    end
+
+    class VCenterNoTagFoundError < StandardError
+      def initialize(category_name)
+        @category_name = category_name
+      end
+
+      def to_s
+        "No tags in category '#{@category_name}' exist on vCenter, skip tagging this category to vm."
```

Comment:
Message can be better `None of the provided tags exist for the category: #{@category_name} on vCenter`

Reply:
Done.